test/CMakeLists.txt

```diff
@@ -15,13 +15,19 @@ if(SPIRV_TOOLS_FOUND)
     get_filename_component(SPIRV_TOOLS_BINDIR "${SPIRV_TOOLS_SPIRV_VAL}" DIRECTORY)
     set(SPIRV_TOOLS_SPIRV_VAL_FOUND True)
   endif()
+
+  find_program(SPIRV_TOOLS_SPIRV_AS NAMES spirv-as PATHS ${SPIRV_TOOLS_PREFIX}/bin)
 endif()
 
 if(NOT SPIRV_TOOLS_SPIRV_VAL)
   message(WARNING "spirv-val not found! SPIR-V generated for test suite will not be validated.")
   set(SPIRV_TOOLS_SPIRV_VAL_FOUND False)
 endif()
 
+if(NOT SPIRV_TOOLS_SPIRV_AS)
+  message(FATAL_ERROR "spirv-as not found! Needed for SPIR-V assembly tests.")
```

Comment:
I think this has been discussed already [in this thread](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/180#discussion_r283801824) - having hard dependency on spirv-tools at least have to be explicitly communicated to downstream users.

I think that it is okay to have it once we really use SPIR-V text representations from the spirv-tools, but for now it looks quite strange/restrictive that this dependency is needed for nothing more than a few tests.

Tagging @AlexeySotkin and @svenvh for more opinions

Reply:
I could alter the dependency so that a warning is produced instead when `spirv-as` cannot be found. `config.suffixes` in `test/lit.cfg.py` would only be updated to include `'.spvasm'` if `spirv-as` was found. `README.md` would need to be updated to mention this in the _Test instructions_ section alongside the existing explanation of `spirv-val`. Would this be okay?

Reply:
Not sure that having some tests that are intended to test core features, but disabled due to missed optional dependency is a good idea.

Probably if all our tests used spirv-tools we could claim that `spirv-as`/`spirv-dis` tools are required to perform testing and make hard dependency on the spirv-tools. But I think that we need to discuss this with the rest of the translator community.

@AnastasiaStulova, @AlexeySotkin, @svenvh, we have faced this discussion second time already (previously [here](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/180#discussion_r283801824)), would it worth adding this topic into agenda for tooling call?

Reply:
> @AnastasiaStulova, @AlexeySotkin, @svenvh, we have faced this discussion second time already (previously [here](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/180#discussion_r283801824)), would it worth adding this topic into agenda for tooling call?

I agree that we should discuss this in the tooling call before merging.

Reply:
Yes, a warning is definitely better than a fatal error.
But also I'd prefer to add something like `; REQUIRE: spirv-as` to the test and update the  test config to enable them if spirv-as is found. This way the user will explicitly see which tests are skipped as UNSUPPORTED.

Looking forward, I think text representation of SPIR-V should be consistent across all tests for the translator. It is worth discussing if we want to take direction to switching SPIR-V text representation to the spirv-tools format for all tests.

Reply:
Okay. I have pushed a new version that adds `spirv-as` as an optional dependency, and adds `; REQUIRES: spirv-as` to each of the SPIR-V assembly tests. I have confirmed that they are listed as UNSUPPORTED when running without `spirv-as`.

Reply:
@StuartDBrady, I, personally, okay to merge this since these tests will be launched by Travis CI and downstream users are responsible to configure environment correctly if they want to perform full testing of the translator. Of course, it still looks a little bit strange - tests for core feature, but they are effectively optional.

It would be really nice to see a corresponding issue to continue this discussion if this patch gets merged

Reply:
@AlexeySachkov, we do have an issue for switching to the SPIR-V assembly format, which is issue #211. As more translator tests get added in SPIR-V assembly format (and converted from the internal SPIR-V text format), I imagine we might want to make running of the test suite conditional upon having `spirv-as` and `spirv-val`. This would be done without adding a dependency for users building the translator without wanting to run the tests.

If this revision is acceptable, I suggest we continue discussion on issue #211.